mousestyles/path_diversity/detect_noise.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import print_function, absolute_import, division
+
+import pandas as pd
+from path_features import compute_angles
+
+
+def detect_noise(movement, paths, angle_threshold, delta_t):
+    r"""
+    Return a list object containing boolean values at points
+    where measurement noise is detected and will be passed to
+    a smoothing function
+
+    Parameters
+    ----------
+    movement : pandas.DataFrame
+        CT, CX, CY coordinates and homebase status
+        for the unique combination of strain, mouse and day
+
+    paths index : a list containing the indices for all paths
+
+    angle_threshold : float
+        positive number indicating the minimum turning angle to flag as noise
+
+    delta_t : float
+        positive number indicating the delta_time interval
+
+    Returns
+    -------
+    noise index : a pandas series containing the indices at which
+    noise, as defined by input parameters, is detected
+
+    Examples
+    --------
+    >>> movement = data.load_movement(1, 2, 1)
+    >>> paths = path_index(movement, 1, 1)
+    >>> noise = detect_noise(movement, paths, 135, .1)
+    """
+
+    # check if all inputs are positive
+    conditions_value = [angle_threshold < 0, delta_t < 0]
+    if any(conditions_value):
+        raise ValueError("Input values need to be positive")
+
+    if not isinstance(movement, pd.core.frame.DataFrame):
+        raise TypeError("Movement must be pandas DataFrame")
+
+    if set(movement.keys()) != {'isHB', 't', 'x', 'y'}:
+        raise ValueError(
+            "The keys of movement must be 't', 'x', 'y', and 'isHB'")
+
+    if len(movement) <= 1:
+        raise ValueError("Movement must contain at least 2 rows")
+
+    noise_index = 1
+    noise_path = []
+    noise_path = pd.Series(noise_path)
+    current_noise = False
+
+    for path in paths:
+        path_obj = movement[path[0]:path[1] + 1]
+
+        if len(path_obj) > 3:
+
+            path_obj['angles'] = compute_angles(path_obj, False)
+            path_obj['sharp_angle'] = path_obj['angles'] > angle_threshold
+            path_obj['noise'] = 0
+
```

Comment:
I found that `insert` would works well when you add new columns to the data frame. For example you could do like
`path_obj.insert(loc = len(one_path.columns), column = 'angles', value = compute_angles(one_path, False))
path_obj.insert(loc = len(one_path.columns), column = 'angles', value = compute_angles(one_path, False))`